### DISCUSSION TITLE
Vercel custom domain not appearing in search engines

### DISCUSSION THREAD

**Discussion by thanosoncode** [vercel/community#3726] [Help]
### Summary

I bought a domain name from Vercel almost a month ago and linked it to a project I have also deployed on Vercel. Although the website is up and running with that domain name it does not appear in any search engine on any page. 

DOMAIN NAME: atlaschania.gr
REGISTRAR: Vercel
NAMESERVERS: Vercel
Edge Network: Active
Age: 26d

Editing Google maps and adding the website to the pin was accepted by them by the way. 

Are there any additional steps to be done for the website searchable? (sitemap and metadata added)

I am not asking why the site does not come first on google, I am asking why it does not come up at all. 
It is like it is still using a subdomain .vercel.app but it is not.
 
 Any help?
 

### Steps to Reproduce

1. Visit atlaschania.gr (website is using a custom domain and not a subdomain on .vercel.app) 

2. Search "atlas chania" on google (Google maps widget includes the website as a valid link) 

3. Go through the search results (the website name does not appear in any of the 3 pages of the results)

**Comment by amyegan** [comment]
Hey there. I'm not an SEO expert, so I can't really offer specific advice on this. But I'd recommend checking some resources, like the [Search Engine Optimization (SEO) Starter Guide](https://developers.google.com/search/docs/fundamentals/seo-starter-guide?visit_id=638284200634051085-1128678163&rd=1) from Google, [The Beginner's Guide to SEO](https://moz.com/beginners-guide-to-seo) from Moz, or [The Ultimate Guide to SEO in 2023](https://blog.hubspot.com/marketing/seo) from HubSpot. They all have tips about site setup that may help.

It's also a good idea to [ask Google to recrawl your URLs](https://developers.google.com/search/docs/crawling-indexing/ask-google-to-recrawl) since you've made a domain change. That may end up being the solution.

**Comment by thanosoncode** [comment]
Thank you for the links @amyegan 

[Sitemap](https://www.atlaschania.gr/sitemap.xml) was accepted by search.google.com but **indexing** failed (after 2 days of pending) due to **redirecting**. I guess the issue is that I requested indexing for atlaschania.gr and not www.atlaschania.gr. I repeat the whole process for the latter and see how it goes 🤞

**Reply by louishugens** [reply]
Hey @thanosoncode,

How is it going? I am experiencing the same. It seems Google crawler is choosing the vercel.app domain as canonical and is in conflict with my chosen canonical ending up  in indexing failure. I'm still looking for a way to fix this without success so far. If anything on your hand would you please share?
Thanks

**Reply by thanosoncode** [reply]
Hello @louishugens,

If you're under vercel.app you can not be indexed by Google ever, you need to have you own custom domain. 

To set my custom domain as canonical I added this in the `Head` in the `layout.tsx `and after 1-2 weeks it was picked up from Google

 `<link rel="canonical" href="https://www.atlaschania.gr/" />`
  
If you're not using Next you just add it like this
  
```
<html>
<head>
<title>Some title</title>
<link rel="canonical" href="https://example.com" />
<!-- other elements -->
</head>
<!-- rest of the HTML -->
```

**Reply by darenminarolli** [reply]
Hey also I wonder did next js helped with the seo ?
